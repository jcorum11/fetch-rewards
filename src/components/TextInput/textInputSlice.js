import { createSlice } from '@reduxjs/toolkit';

const textInputSlice = createSlice({
  name: 'textInput',
  initialState: {
    name: {
      first: '',
      middle: '',
      last: ''
    },
    email: '',
    password: ''
  },
  reducers: {
    setFirstName: (state, action) => {
      state.name.first = action.payload
    },
    setMiddleName: (state, action) => {
      state.name.middle = action.payload
    },
    setLastName: (state, action) => {
      state.name.last = action.payload
    },
    setName: (state, action) => {
      state.name = action.payload
    },
    setEmail: (state, action) => {
      state.email = action.payload
    },
    setPassword: (state, action) => {
      state.password = action.payload
    }
  }
})

export const selectName = state => state.textInput.name
export const selectEmail = state => state.textInput.email
export const selectPassword = state => state.textInput.password

export const { setFirstName, setMiddleName, setLastName, setName, setEmail, setPassword } = textInputSlice.actions

export default textInputSlice.reducer