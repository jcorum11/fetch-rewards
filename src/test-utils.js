import React from 'react'
import { render as rtlRender } from '@testing-library/react'
import { configureStore } from '@reduxjs/toolkit'
import { Provider } from 'react-redux'
// Import your own reducer
import dropdownReducer from './components/DropdownInput/dropdownInputSlice'
import textInputReducer from './components/TextInput/textInputSlice'
import appReducer from './pages/App/appSlice'

function render(
  ui,
  {
    preloadedState,
    store = configureStore({ reducer: { dropdown: dropdownReducer, textInput: textInputReducer, app: appReducer }, preloadedState }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react'
// override render method
export { render }