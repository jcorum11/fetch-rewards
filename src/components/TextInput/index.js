/** @jsxImportSource @emotion/react */
import { useState } from 'react'
import { useDispatch } from 'react-redux'
import { setEmail, setFirstName, setLastName, setMiddleName, setPassword } from './textInputSlice'
import { css } from '@emotion/react'

const TextInput = ({ type, nameType = '' }) => {
  const dispatch = useDispatch()
  const [value, setValue] = useState('')
  const handleChange = (event) => {
    setValue(event.target.value)
    switch (type) {
      case 'Name':
        if (nameType === 'First') {
          dispatch(setFirstName(event.target.value))
        } else if (nameType === 'Middle') {
          dispatch(setMiddleName(event.target.value))
        } else if (nameType === 'Last') {
          dispatch(setLastName(event.target.value))
        }
        break
      case 'Password':
        dispatch(setPassword(event.target.value))
        break
      case 'Email':
        dispatch(setEmail(event.target.value))
        break
      default:
        break
    }
  }
  const cleanType = type === 'name' ? `${nameType}input`.toLowerCase() : `${type}input`.toLowerCase()
  return (
    <div css={inputContainer}>
      <label css={label} htmlFor={cleanType} data-testid={`${nameType.toLowerCase()}${cleanType}-label`}>{nameType} {type}: </label>
      <input css={input} type="text" name={type.toLowerCase()} id={cleanType} data-testid={`${nameType.toLowerCase()}${cleanType}-input`} value={value} onChange={handleChange} />
    </div>
  )
}

const label = css`
display: block;
color: white;
`

const inputContainer = css`
padding: .5rem;
`

const input = css`
width: 98%;
height: 1.5rem;
border-radius: 5px;
`

export default TextInput
