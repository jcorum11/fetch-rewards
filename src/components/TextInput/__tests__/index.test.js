import React from 'react';
import TextInput from '..';
import {render, cleanup} from '../../../test-utils'
import '@testing-library/jest-dom/extend-expect'

afterEach(cleanup)

describe('TextInput', () => {
  it('Renders', async () => {
    render(<TextInput type='Name' nameType='First' />)
  })

  it('Matches snapshot', async () => {
    const { asFragment } = render(<TextInput type='Name' nameType='First' />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('Inserts text into label', () => {
    const { getByTestId } = render(<TextInput type='Name' nameType='First' />)
    expect(getByTestId('firstnameinput-label')).toHaveTextContent('First Name:')
  })
})
