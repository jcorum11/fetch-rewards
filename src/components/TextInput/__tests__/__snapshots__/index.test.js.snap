// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput Matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1yuj46c-TextInput"
  >
    <label
      class="css-5e23jr-TextInput"
      data-testid="firstnameinput-label"
      for="nameinput"
    >
      First Name: 
    </label>
    <input
      class="css-1a9jkjf-TextInput"
      data-testid="firstnameinput-input"
      id="nameinput"
      name="name"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
